db/schema.rb

```diff
@@ -10,11 +10,23 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20180125202414) do
+ActiveRecord::Schema.define(version: 20180131214527) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "submissions", force: :cascade do |t|
+    t.string "title"
+    t.string "url"
+    t.text "text"
+    t.bigint "user_id", null: false
+    t.bigint "post_id"
```

Comment:
Does a null here map to a submission that just asks a question to the community?

Reply:
The null here is operating at the data table level to require that a `user_id` be present.

Are you asking if a user can post a question to the community to start a discussion thread? Someone could use the the "text" post option for that. Maybe no one makes that connection and that's why there are so few "text" posts on HN.

Reply:
Ah I was referring to the `post_id`. Cool. So a `text` type post has no post. 👍

Reply:
@nhunzaker -- yes, the idea is that if `post_id` is `null`, then a submission should be treated as a "post" vs. a "comment".

Reply:
@nhunzaker - sorry for not getting what you were asking. Part of the confusion may be because I wrote validations for Submissions with posts in mind, then added the comment functionality, but haven't gone back to update the validations.

Here's a break down of the options:
**A Submission can be a post OR a comment**
 * If post_id is null, the Submission is a post;
 * If post_id is not null, the Submission is a comment

**When submitting a post, a User can submit either**
 * a url and title OR, 
 * text content

**When submitting a comment, a user submits**
* text content

*Functionality that hasn't yet been written*
**Comments can be:**
* In response to posts
* Replies to other comments

As I wrote in the PR description, the advantage to this approach is that one table holds all submissions. There could be disadvantages but I haven’t run into any . . . yet :)